atlassian/bitbucket/cloud/repositories/pullRequests.py

```diff
@@ -0,0 +1,169 @@
+# coding=utf-8
+
+from ..base import BitbucketCloudBase
+from .defaultReviewers import DefaultReviewer
+from datetime import datetime
+
+
+class PullRequests(BitbucketCloudBase):
+    def __init__(self, url, *args, **kwargs):
+        super(PullRequests, self).__init__(url, *args, **kwargs)
+
+    def __get_object(self, data):
+        if "errors" in data:
+            return
+        return PullRequest(self.url_joiner(self.url, data["id"]), data, **self._new_session_args)
+
+    def each(self, q=None, sort=None):
+        """
+        Returns the list of pull requests in this repository.
+
+        :param q: string: Query string to narrow down the response.
+                          See https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering for details.
+        :param sort: string: Name of a response property to sort results.
+                             See https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering for details.
+
+        :return: A generator for the PullRequest objects
+        """
+        params = {}
+        if sort is not None:
+            params["sort"] = sort
+        if q is not None:
+            params["q"] = q
+        for pr in self._get_paged(None, trailing=True, params=params):
+            yield self.__get_object(pr)
+
+        return
+
+    def get(self, id):
+        """
+        Returns the pull requests with the requested id in this repository.
+
+        :param id: int: The requested pull request id
+
+        :return: The requested PullRequest object
+        """
+        return self.__get_object(super(PullRequests, self).get(id))
+
+
+class PullRequest(BitbucketCloudBase):
+    def __init__(self, url, data, *args, **kwargs):
+        super(PullRequest, self).__init__(url, *args, data=data, expected_type="pullrequest", **kwargs)
+
+    @property
+    def id(self):
+        """ unique pull request id """
+        return self.get_data("id")
+
+    @property
+    def title(self):
+        """ pull request title """
+        return self.get_data("title")
+
+    @property
+    def description(self):
+        """ pull request description """
+        return self.get_data("description")
+
+    @property
+    def state(self):
+        """
+        pull request state
+        possible values: MERGED, SUPERSEDED, OPEN, DECLINED
+        """
+        return self.get_data("state")
```

Comment:
Is it better to add a property for each state (`is_merged`, `is_superseded`, `is_open` or `is_declined`) instead of or in addition to returning the strings from BB?

Reply:
sure, thats way better. Will add this..